payloads/v9/_interactions/applicationCommands.ts

```diff
@@ -267,16 +310,44 @@ export const enum ApplicationCommandPermissionType {
 
 // INTERACTIONS
 
-export type APIApplicationCommandInteraction = APIBaseInteraction<
-	InteractionType.ApplicationCommand,
-	APIApplicationCommandInteractionData
-> &
-	Required<
-		Pick<
-			APIBaseInteraction<InteractionType.ApplicationCommand, APIApplicationCommandInteractionData>,
-			'channel_id' | 'data'
-		>
-	>;
+export type APIApplicationCommandInteractionWrapper<Data extends APIApplicationCommandInteractionData> =
+	APIBaseInteraction<InteractionType.ApplicationCommand, Data> &
+		Required<Pick<APIBaseInteraction<InteractionType.ApplicationCommand, Data>, 'channel_id' | 'data'>>;
+
+export type APIChatInputApplicationCommandInteraction =
+	APIApplicationCommandInteractionWrapper<APIChatInputApplicationCommandInteractionData>;
+
+export type APIChatInputApplicationCommandDMInteraction =
+	APIDMInteractionWrapper<APIChatInputApplicationCommandInteraction>;
+
+export type APIChatInputApplicationCommandGuildInteraction =
+	APIGuildInteractionWrapper<APIChatInputApplicationCommandInteraction>;
+
+export type APIUserApplicationCommandInteraction =
+	APIApplicationCommandInteractionWrapper<APIUserApplicationCommandInteractionData>;
+
+export type APIUserApplicationCommandDMInteraction = APIDMInteractionWrapper<APIUserApplicationCommandInteraction>;
+
+export type APIUserApplicationCommandGuildInteraction =
+	APIGuildInteractionWrapper<APIUserApplicationCommandInteraction>;
+
+export type APIMessageApplicationCommandInteraction =
+	APIApplicationCommandInteractionWrapper<APIMessageApplicationCommandInteractionData>;
+
+export type APIMessageApplicationCommandDMInteraction =
+	APIDMInteractionWrapper<APIMessageApplicationCommandInteraction>;
+
+export type APIMessageApplicationCommandGuildInteraction =
+	APIGuildInteractionWrapper<APIMessageApplicationCommandInteraction>;
+
+export type APIContextMenuInteraction = APIApplicationCommandInteractionWrapper<APIContextMenuInteractionData>;
+
+export type APIContextMenuDMInteraction = APIDMInteractionWrapper<APIContextMenuInteraction>;
+
+export type APIContextMenuGuildInteraction = APIGuildInteractionWrapper<APIContextMenuInteraction>;
+
+export type APIApplicationCommandInteraction =
```

Comment:
shouldn't this be a union of the previous ones (the context menu and chat input ones)?